nov.el

```diff
@@ -742,5 +744,41 @@ Saving is only done if `nov-save-place-file' is set."
 (add-hook 'nov-mode-hook 'nov-add-to-recentf)
 (add-hook 'nov-mode-hook 'hack-dir-local-variables-non-file-buffer)
 
+
+;; Bookmark Integration
+(defun nov-bookmark-make-record  ()
+  "Create a bookmark epub record."
+  (cons (buffer-name)
+        `((filename . ,nov-file-name)
+          (index . ,nov-documents-index)
+          (position . ,(point))
+          (handler . nov-bookmark-jump-handler))))
+
+;;;###autoload
+(defun nov-bookmark-jump-handler (bmk)
+  "The bookmark handler-function interface for bookmark BMK.
+
+See also `nov-bookmark-make-record'."
+  (let ((file (bookmark-prop-get bmk 'filename))
+        (index (bookmark-prop-get bmk 'index))
+        (position (bookmark-prop-get bmk 'position)))
+    (switch-to-buffer (or (cl-find-if (lambda (buffer)
```

Comment:
This is needlessly complicated. All you need is `(find-file file)`, it will switch to the buffer if it exists or create one if it doesn't.